app/ai/voice/agents/breeze_buddy/services/rag/slow_thinker.py

```
@@ -0,0 +1,264 @@
+                if now - self._last_run < self._rate_limit:
+                    continue  # Rate-limit: skip this cycle
+                self._last_run = now
+
+                if kind == "utterance":
+                    await self._handle_utterance(text, history)
+
+            except asyncio.TimeoutError:
+                continue
+            except asyncio.CancelledError:
+                break
+            except Exception as exc:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)